lib/engine/step/route.rb

```diff
@@ -41,6 +41,10 @@ def process_run_routes(action)
           @log << "#{entity.name} runs a #{train.name} train for #{revenue}: #{route.revenue_str}"
         end
         pass!
+
+        return unless (route = @round.routes.first)
```

Comment:
this doesn't look correct? why do you only use one route?

Reply:
why not just put it in the each loop ?

Reply:
Changed this code a bit.